### ISSUE TITLE
[Discussion]: all instances of new Job should be replaced with JobMaker.MakeJob

### ISSUE THREAD

**Issue by MadaraUchiha** [CombatExtended-Continued/CombatExtended#96]
I don't currently have the time to do this, so putting it here for future Madara or other team members to pick up.

All instances of `new Job()` we have should be replaced with `JobMaker.MakeJob()` for performance and memory reasons. `JobMaker.MakerJob` uses an internal job pool to prevent object allocation and other goodies. They use the same signature.

**Comment by Dr-Slump** [comment]
Something like that?

https://github.com/Dr-Slump/CombatExtended/commit/51f96311a99df3baa92cde7595295b74c51ab265

**Comment by MadaraUchiha** [comment]
@Dr-Slump Yes, exactly like that, actually. Care to open a pull request? :)
